### PR TITLE
Add getAllStateVariablesPaths

### PR DESCRIPTION
**Pull request by PavelBal**

### CODE REVIEW

File: R/utilities-simulation.R [hunk 1]
```diff
@@ -425,6 +425,22 @@ getAllParametersForSensitivityAnalysisMatching <- function(paths, simulation) {
...
+#' Get the paths of all state variable quantities of the simulation
+#'
+#' @param simulation \code{Simulation} object
+#' @details List of paths of all molecules in all compartments and all parameters that are
+#' state variables.
+#'
+#' @return A list of paths
+#' @export
+getAllStateVariablesPaths <- function(simulation) {
+  validateIsOfType(simulation, type = "Simulation")
```
Comment (msevestre) [inline R/utilities-simulation.R:437]: you can pass a ref to the R6 class directly here (e..g validateIsOfType(simulation, Simulation) which removes the use of "magic" string

File: R/utilities-simulation.R [hunk 2]
```diff
@@ -425,6 +425,22 @@ getAllParametersForSensitivityAnalysisMatching <- function(paths, simulation) {
...
+  allMoleculesPaths <- getAllEntityPathsIn(container = simulation, entityType = Molecule)
```
Comment (msevestre) [inline R/utilities-simulation.R:438]: getAllMoleculePathsIn instead. Since we have a convenience method for this, why not use it?

**Review by msevestre — CHANGES_REQUESTED** [review]

File: R/utilities-simulation.R (hunk 2 above)
Comment (PavelBal) [reply R/utilities-simulation.R]: Done

File: R/utilities-simulation.R (hunk 1 above)
Comment (PavelBal) [reply R/utilities-simulation.R]: Done

**Pull request merged**